pkg/informant/endpoints.go

```diff
@@ -258,7 +258,22 @@ func (s *State) TryDownscale(ctx context.Context, target *api.RawResources) (*ap
 		return resultFromStatus(true, "No action taken (no cgroup or file cache enabled)")
 	}
 
-	requestedMem := uint64(target.Memory.Value())
+	currentId := s.agents.current.id
+	incomingId := target.Data.Id.AgentID
+	// This condition deals with two cases:
+	// 1: The agent that responded is legitimately an unknown agent
+	// 2: The agent that responded is using an old protocol, and thus did not respond with an AgentId.
+	//    In this case, due to Go's uninitialized variable semantics, the "AgentID" we read will be
+	//    all 0's, which will never match, since UUID's cannot be all 0's
+	if incomingId != currentId {
+		klog.Errorf(
+			"Got downscale response from agent %v, while current agent is %v",
+			incomingId, currentId,
+		)
+		return nil, 400, fmt.Errorf("Received downscale response from unknown agent %v", incomingId)
+	}
```

Comment:
Missed this in earlier comments, but similar "move this above the other check" for the same reason.